src/base_client.rs

```diff
@@ -209,12 +209,6 @@ impl Client {
             *olm = Some(OlmMachine::new(&response.user_id, &response.device_id)?);
         }
 
-        if let Some(store) = self.state_store.as_ref() {
-            store
-                .store_client_state(ClientState::from_base_client(self))
-                .await?;
-        }
```

Comment:
I realize this overwrites `ClientState` when there is no state other than `UserId` and `DeviceId` to save so instead added the logic to append to the `JsonStore` path when `store_room_state` is called.